### PR TITLE
Ceil op for TFLITE

### PR DESCRIPTION
**Pull request by siju-samuel**
CEIL basic ops for tensorflow lite

### CODE REVIEW

**Comment by siju-samuel** [comment]
@aselle could you please help to review this ops. TIA

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @aselle: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/lite/delegates/nnapi/nnapi_delegate.cc [hunk 1]
```diff
@@ -638,6 +638,16 @@ class NNAPIDelegateKernel {
           return nullptr;
         }
         break;
+      case kTfLiteBuiltinCeil:
+        if (version == 1) {
+          return [](const NNAPIOpMappingArgs& mapping_args)
+                     -> ANeuralNetworksOperationType {
+            return ANEURALNETWORKS_CEIL;
```
Comment (jdduke) [inline tensorflow/lite/delegates/nnapi/nnapi_delegate.cc]: Does this NNAPI op exist?

File: tensorflow/lite/kernels/BUILD [hunk 2]
```diff
@@ -599,6 +600,22 @@ tf_cc_test(
     ],
 )
 
+tf_cc_test(
+    name = "ceil_test",
+    size = "small",
+    srcs = ["ceil_test.cc"],
+    tags = [
+        "no_oss",
```
Comment (jdduke) [inline tensorflow/lite/kernels/BUILD]: Remove "no_oss", this should be fine

File: tensorflow/lite/kernels/internal/optimized/legacy_optimized_ops.h [hunk 3]
```diff
@@ -1740,6 +1740,12 @@ inline void Floor(const float* input_data, const Dims<4>& input_dims,
         output_data);
 }
 
+inline void Ceil(const float* input_data, const Dims<4>& input_dims,
```
Comment (jdduke) [inline tensorflow/lite/kernels/internal/optimized/legacy_optimized_ops.h]: I thin you can revert the change to this file, it shouldn't be necessary.

File: tensorflow/lite/kernels/internal/reference/legacy_reference_ops.h [hunk 4]
```diff
@@ -1883,6 +1883,12 @@ inline void Floor(const float* input_data, const Dims<4>& input_dims,
         output_data);
 }
 
+inline void Ceil(const float* input_data, const Dims<4>& input_dims,
```
Comment (jdduke) [inline tensorflow/lite/kernels/internal/reference/legacy_reference_ops.h]: Same here, addition to this file should no longer be necessary.

File: tensorflow/lite/nnapi/NeuralNetworksShim.h [hunk 5]
```diff
@@ -143,6 +143,7 @@ enum {
   ANEURALNETWORKS_STRIDED_SLICE = 35,
   ANEURALNETWORKS_SUB = 36,
   ANEURALNETWORKS_TRANSPOSE = 37,
+  ANEURALNETWORKS_CEIL = 38,
```
Comment (jdduke) [inline tensorflow/lite/nnapi/NeuralNetworksShim.h]: This code isn't in Android so I don't think we can land it. Can we factor out the NNAPI additions separately?

**Review by jdduke — COMMENTED** [review]
Looks really good, thanks for the addition! Just a few minor comments.

**Comment by siju-samuel** [comment]
@jdduke Thank you very much for your time and reviewing my code. I have updated the code based on your review comments.

**Comment by siju-samuel** [comment]
@jdduke @aselle @dksb Could you please review again and approve if everything is fine? thanks in advance.

**Review by jdduke — DISMISSED** [review]
Looks good, thanks!

**Comment by hgadig** [comment]
@siju-samuel  Can you please run clang-format on your code.

**Comment by siju-samuel** [comment]
@jdduke please approve again. just updated some spacing issues found using clang-format.

@hgadig the sanity test was failing due to the below error

```
=== Sanity check step 5 of 12: do_bazel_nobuild (bazel nobuild) ===
FAIL: bazel build --nobuild  -- //tensorflow/... -//tensorflow/lite/java/demo/app/... -//tensorflow/lite/examples/android/... -//tensorflow/lite/schema/...
  This is due to invalid BUILD files. See lines above for details.
ERROR: error loading package 'tensorflow/lite/experimental/swift': Extension file not found. Unable to load file '@bazel_skylib//lib:partial.bzl': file doesn't exist or isn't a file
```

**Review by jdduke — APPROVED** [review]

**Comment by siju-samuel** [comment]
@hgadig Anything pending from my side for this PR?

**Comment by hgadig** [comment]
> @hgadig Anything pending from my side for this PR?

Nothing from your end. We are looking into this.

**Comment by jdduke** [comment]
There are some minor tweaks we need to make internally, it should land in the next day or so, thanks for your patience.
